src/Akka.Persistence.Hosting/AkkaPersistenceHostingExtensions.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Akka.Configuration;
+using Akka.Hosting;
+using Akka.Persistence.Journal;
+using Akka.Util;
+
+namespace Akka.Persistence.Hosting
+{
+    public enum PersistenceMode
+    {
+        /// <summary>
+        /// Sets both the akka.persistence.journal and the akka.persistence.snapshot-store to use this plugin.
+        /// </summary>
+        Both,
+
+        /// <summary>
+        /// Sets ONLY the akka.persistence.journal to use this plugin.
+        /// </summary>
+        Journal,
+
+        /// <summary>
+        /// Sets ONLY the akka.persistence.snapshot-store to use this plugin.
+        /// </summary>
+        SnapshotStore,
+    }
+    
+    /// <summary>
+    /// Used to help build journal configurations
+    /// </summary>
+    public sealed class AkkaPersistenceJournalBuilder
+    {
+        internal readonly string JournalId;
+        internal readonly AkkaConfigurationBuilder Builder;
+        internal readonly Dictionary<Type, HashSet<string>> Bindings = new Dictionary<Type, HashSet<string>>();
+        internal readonly Dictionary<string, Type> Adapters = new Dictionary<string, Type>();
+
+        public AkkaPersistenceJournalBuilder(string journalId, AkkaConfigurationBuilder builder)
+        {
+            JournalId = journalId;
+            Builder = builder;
+        }
+
+        public AkkaPersistenceJournalBuilder AddEventAdapter<TAdapter>(string eventAdapterName,
```

Comment:
Had to add separate methods for each of the different event adapter interfaces since there isn't a common base one.